syncapi/sync/requestpool.go

```diff
@@ -80,6 +92,52 @@ func (rp *RequestPool) cleanLastSeen() {
 	}
 }
 
+func (rp *RequestPool) cleanPresence(cleanupTime time.Duration) {
+	for {
+		rp.presence.Range(func(key interface{}, v interface{}) bool {
+			p := v.(types.Presence)
+			if time.Since(p.LastActiveTS.Time()) > cleanupTime {
+				rp.updatePresence("unavailable", p.UserID)
+				rp.presence.Delete(key)
+			}
+			return true
+		})
+		time.Sleep(cleanupTime)
+	}
+}
+
+// updatePresence sends presence updates to the SyncAPI and FederationAPI
+func (rp *RequestPool) updatePresence(presence string, userID string) {
+	if rp.cfg.Matrix.DisablePresence {
+		return
+	}
+	if presence == "" {
+		presence = "online"
```

Comment:
If these terms like `"online"`, `"unavailable"` have special meaning from the spec, let's define them as `const` somewhere.